src/Raven.Server/ServerWide/Maintenance/ClusterObserver.cs

```diff
@@ -911,21 +909,24 @@ private void RemoveOtherNodesIfNeeded(DatabaseObservationState state, ref List<D
                 mentorChangeVector.Add(node, dbReport.DatabaseChangeVector);
             }
 
-            if (nodesToDelete.Count > 0)
+            if (nodesToDelete.Count == 0)
+                return;
+
+            LogMessage($"We reached the replication factor on database '{dbName}', so we try to remove redundant nodes from {string.Join(", ", nodesToDelete)}.",
+                info: false);
+
+            var deletionCmd = new DeleteDatabaseCommand
             {
-                var deletionCmd = new DeleteDatabaseCommand
-                {
-                    ErrorOnDatabaseDoesNotExists = false,
-                    DatabaseName = dbName,
-                    FromNodes = nodesToDelete.ToArray(),
-                    HardDelete = _hardDeleteOnReplacement,
-                    UpdateReplicationFactor = false,
-                };
-
-                if (deletions == null)
-                    deletions = new List<DeleteDatabaseCommand>();
-                deletions.Add(deletionCmd);
-            }
+                ErrorOnDatabaseDoesNotExists = false,
+                DatabaseName = dbName,
+                FromNodes = nodesToDelete.ToArray(),
+                HardDelete = _hardDeleteOnReplacement,
+                UpdateReplicationFactor = false,
+            };
+
+            if (deletions == null)
+                deletions = new List<DeleteDatabaseCommand>();
+            deletions.Add(deletionCmd);
```

Comment:
And here is the code that does that. Does this run only once?

Reply:
It is too frequent to be called in the observer and here we can't tell between `failed to notify` and `deletion in progress`.

On the relevant node we will retry to send the command again after 15 seconds.

Reply:
"too frequent" ? I'm not sure that I follow. Isn't that twice a second?

Reply:
Okay, if there is a retry, that is fine